File: sandbox/copy_with_linenum.rb
#! ruby -Ks
# -*- coding: Shift_JIS -*-

=begin
比較用Ruby
コマンドライン引数で渡されたテキストファイル「ほにゃ.uga」を読み込み、先頭に行番号を付加した「ほにゃ.uga.with_lineno.txt」を生成する
Test / Ruby1.8.7
=end

open ARGV[0] do |src|
	open "#{ARGV[0]}.with_lineno.txt", 'w' do |dst|
		src.each_with_index do |line, i|
			dst << "#{i + 1}:#{line}"
		end
	end
end

File: sandbox/copy_with_linenum.lua
#! lua

--[[
コマンドライン引数で渡されたテキストファイル「ほにゃ.uga」を読み込み、先頭に行番号を付加した「ほにゃ.uga.with_lineno.txt」を生成する
Test / Lua 5.1.3
--]]

io.output(arg[1]..'.with_lineno.txt')

lineNum = 0

for line in io.lines(arg[1]) do
	lineNum = lineNum + 1
	io.write(string.format("%d:", lineNum)..line.."\n")
end